File: Implementacion/inteligencia/sinMemo/casilla.py

# **************************************************************************
#			  casilla.py  -  description
#   	                      ------------------
# Wizard's Sword
# desarrollado por     : (C) 2004 by M Beatriz Medina Yanez
# *************************************************************************


#  Clase que implementa los atributos de una casilla del juego

# tipo:     Tipo de casilla
#           0 -> Hierba
#           1 -> Tierra
#           2 -> Piedra
#           3 -> Poza de agua
#           4 -> Interior casas
#           5 -> Interior castillos	 

# posx:     Posicion x de la casilla
# posy:     Posicion y de la casilla

# nAtr:     Numero de atributos del objeto Casilla

class Casilla:
    __tipo = None
    __posx = None
    __posy = None
    __nAtr = None
    
    # Inicializador
    def __init__ (self):
        self.__nAtr = 3

    # Metodo consultor que devuelve el numero de atributos del objeto Casilla
    def obtieneNAtributos (self):
        return self.__nAtr
    
    # Metodo consultor que devuelve el valor del atributo especificado
    def obtieneAtributo (self, atributo):

        if ((atributo == 1) or (atributo == 'tipo')):
            return self.__tipo

        elif ((atributo == 2) or (atributo == 'posx')):
            return self.__posx

        elif ((atributo == 3) or (atributo == 'posy')):
            return self.__posy

    # Metodo que actualiza con un nuevo valor al atributo especificado    
    def reAsignaAtributo (self, atributo, nuevoValor):

        if ((atributo == 1) or (atributo == 'tipo')):
            self.__tipo = nuevoValor

        elif ((atributo == 2) or (atributo == 'posx')):
            self.__posx = nuevoValor
            
        elif ((atributo == 3) or (atributo == 'posy')):
            self.__posy = nuevoValor

# *************************************************************************

File: Implementacion/casilla.cc

/***************************************************************************
                          casilla.cc  -  description
                             -------------------
  Wizard's Sword
  desarrollado por     : (C) 2004 by Mª Beatriz Medina Yañez
 ***************************************************************************/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "casilla.hh"

/***************************************************************************/

/* Constructor del objeto Casilla que inicializa todos sus atributos a 
   VACIO */
Casilla::Casilla ()
{
    // Inicialización de los atributos a VACIO
    this->tipo = VACIO;
    this->posx = VACIO;
    this->posy = VACIO;

    // Inicialización del número de atributos
    this->nAtr = 3;
}

/***************************************************************************/

// Destructor del objeto Casilla
Casilla::~Casilla ()
{}

/***************************************************************************/

// Método consultor que devuelve el número de atributos del objeto Casilla
int Casilla::obtieneNAtributos (void)
{
    return this->nAtr;
}

/***************************************************************************/

/* Método consultor que devuelve el valor del atributo cuyo nombre se 
   especifica */ 
int Casilla::obtieneAtributo (char *atributo)
{
    if (!strcmp (atributo, "tipo"))
	return this->tipo;

    else if (!strcmp (atributo, "posx"))
	return this->posx;

    else if (!strcmp (atributo, "posy"))
	return this->posy;

    else
    {
	fprintf (stderr, "casilla.cc::obtieneAtributo (char *): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }
}

/***************************************************************************/

/* Método consultor que devuelve el valor del atributo cuyo índice se 
   especifica */ 
int Casilla::obtieneAtributo (int atributo)
{
    if (atributo == 1)
	return this->tipo;

    else if (atributo == 2)
	return this->posx;

    else if (atributo == 3)
	return this->posy;

    else
    {
	fprintf (stderr, "casilla.cc::obtieneAtributo (char *): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }
}

/***************************************************************************/

/* Método que actualiza con un nuevo valor al atributo cuyo nombre se
   especifica */
void Casilla::reAsignaAtributo (char *atributo, int nuevoValor)
{
    if (!strcmp (atributo, "tipo"))
	this->tipo = nuevoValor;
    
    else if (!strcmp (atributo, "posx"))
	this->posx = nuevoValor;

    else if (!strcmp (atributo, "posy"))
	this->posy = nuevoValor;

    else
    {
	fprintf (stderr, "casilla.cc::reAsignaAtributo (char *, int): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }    
}

/***************************************************************************/

/* Método que actualiza con un nuevo valor al atributo cuyo índice se
   especifica */
void Casilla::reAsignaAtributo (int atributo, int nuevoValor)
{
    if (atributo == 1)
	this->tipo = nuevoValor;
    
    else if (atributo == 2)
	this->posx = nuevoValor;

    else if (atributo == 3)
	this->posy = nuevoValor;

    else
    {
	fprintf (stderr, "casilla.cc::reAsignaAtributo (char *, int): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }    
}

/***************************************************************************/

